app/concepts/reservations/use_cases/create_offline.rb

```diff
@@ -0,0 +1,21 @@
+module Reservations
+  module UseCases
+    class CreateOffline
+      attr_reader :repository
+
+      def initialize(repository: Reservations::Repository.new)
+        @repository = repository
+      end
+
+      def call(params:)
+       
+          Reservation.transaction do
+            reservation_params = params.except(:tickets)
+            @reservation = repository.create(reservation_params)
+            Tickets::UseCases::Create.new(reservation: @reservation, tickets: params[:tickets]).call
+          end
```

Comment:
That looks ok, if something went wrong with ticket creation we want to roll back the created reservation 👍 
I see that this logic is common for `CreateOffline` and `CreateOnline`, so maybe you can take this to a separate class like `CreateReservationWithTickets` and call it here and in `CreateOnline`.

Reply:
Could you take a look at a new commit https://github.com/agusiol/montecinema/pull/14/commits/8efa738cface52397d12f67c7a5984f46b8b4847 ?
I am not sure if should repeat `def initialize` ?

Reply:
I checked that commit and it looks like you don't need 3 classes for that. CreateOffline/CreateReservationWithTickets in that case, it's just the same. So we can go with something like this, WDYT?:
```
module Reservations
  module UseCases
    class CreateOffline
      attr_reader :repository

      def initialize(repository: Reservations::Repository.new)
        @repository = repository
      end

      def call(params:)
        Reservation.transaction do
          reservation_params = params.except(:tickets)
          @reservation = @repository.create(reservation_params)
          Tickets::UseCases::Create.new(reservation: @reservation, tickets: params[:tickets]).call
        end
        @reservation
      end
    end
  end
end

module Reservations
  module UseCases
    class CreateOnline
      attr_reader :repository

      def initialize(repository: Reservations::Repository.new)
        @repository = repository
      end

      def call(params:)
        reservation = Reservations::UseCases::CreateOffline.new(params: params)
        send_email
        reservaton
      end

      def send_email
        client = Clients::Repository.new.find_by(@reservation.client_id)
        screening = Screenings::Repository.new.find_by(@reservation.screening_id)
        movie = Movies::Repository.new.find_by(screening.movie_id)
        ReservationMailer.confirmation_email(
          reservation: @reservation,
          email: client.email,
          date: screening.date,
          movie: movie.title
        ).deliver_later
      end
    end
  end
end
```

Reply:
I know that `CreateReservationWithTickets` and `CreateOffline` are exactly the same but I will leave it as it is for now. Maybe I will add some additional functionality to `CreateOffline` . If not I will refactor it later :)